tests/app_tests/endpoint_tests/posts_test.go

```diff
@@ -40,3 +40,40 @@ func TestPostSuccess(t *testing.T) {
 	assert.Equal(t, http.StatusOK, w.Code)
 	assert.Contains(t, w.Body.String(), "TestPost")
 }
+
+func TestPostFailure(t *testing.T) {
+
+	app_settings := common.AppSettings{
+		DatabaseAddress:  "localhost",
+		DatabasePort:     3006,
+		DatabaseUser:     "root",
+		DatabasePassword: "root",
+		DatabaseName:     "urchin",
+		WebserverPort:    8080,
+	}
+
+	database_mock := mocks.DatabaseMock{
+		GetPostHandler: func(post_id int) (common.Post, error) {
+			return common.Post{
+				Title:   "TestPost",
+				Content: "TestContent",
+				Excerpt: "TestExcerpt",
+				Id:      post_id,
+			}, nil
+		},
+	}
+
+	router := app.SetupRoutes(app_settings, database_mock)
+	responseRecorder := httptest.NewRecorder()
+
+	request, err := http.NewRequest("GET", "/post/sampleString", nil)
+
+	if err != nil {
```

Comment:
replace this if statement with `require.Nil(t, err)` - This will assert that the request was created, and if not, the test will fail and not continue.